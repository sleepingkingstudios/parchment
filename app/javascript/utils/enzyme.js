import { valueOrDefault } from './object';
import { safeCapitalize } from './string';

export const extractInjectedComponent = (component) => {
  const injected = component.shallow().find('ComponentWithInjectedProps');

  return injected.exists() ? injected : component;
};

export const toHaveFormControl = expect => (received, expectedControl, withProps) => {
  const formControl = received.find(expectedControl);
  const expectedProps = valueOrDefault(withProps, {});

  expect(formControl).toExist();
  expect(formControl).toHaveProp(expectedProps);

  return { pass: true, message: () => '' };
};

export const toHaveFormGroup = expect => (received, expectedGroup, fn) => {
  const formGroup = received.find(expectedGroup);

  expect(formGroup).toExist();

  fn(extractInjectedComponent(formGroup).shallow());

  return { pass: true, message: () => '' };
};

export const toHaveFormInput = expect => (received, props) => {
  const {
    form,
    inputType,
    prop,
  } = props;

  expect(received).toHaveProp({ prop });

  const formInput = received.find(`${safeCapitalize(prop)}Input`);

  expect(formInput).toExist();
  expect(formInput).toHaveProp({ form });

  const shallowInput = formInput.shallow();

  expect(shallowInput).toExist();
  expect(shallowInput).toHaveDisplayName(valueOrDefault(inputType, 'FormInput'));

  return { pass: true, message: () => '' };
};
