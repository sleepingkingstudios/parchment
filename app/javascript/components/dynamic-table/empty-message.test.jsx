import React from 'react';
import { shallow } from 'enzyme';

import DynamicTableEmptyMessage from './empty-message';

describe('<DynamicTableEmptyMessage />', () => {
  const message = 'There are no widgets matching the criteria.';
  const defaultProps = { message };

  describe('with default props', () => {
    it('should render the table empty message', () => {
      const rendered = shallow(<DynamicTableEmptyMessage {...defaultProps} />);

      expect(rendered).toHaveDisplayName('div');
      expect(rendered).toHaveClassName('dynamic-table-empty-message');
      expect(rendered).toHaveClassName('col-12');
      expect(rendered).toHaveText(message);
    });

    it('should match the snapshot', () => {
      const rendered = shallow(<DynamicTableEmptyMessage {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('with message: an element', () => {
    const element = (<span>Something went wrong.</span>);

    it('should render the table empty message', () => {
      const rendered = shallow(<DynamicTableEmptyMessage {...defaultProps} message={element} />);

      expect(rendered).toHaveDisplayName('div');
      expect(rendered).toHaveClassName('dynamic-table-empty-message');
      expect(rendered).toHaveClassName('col-12');
      expect(rendered).toHaveProp('children', element);
      expect(rendered).toHaveText('Something went wrong.');
    });
  });
});
