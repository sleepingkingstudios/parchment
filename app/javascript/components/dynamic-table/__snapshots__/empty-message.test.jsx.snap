// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DynamicTableEmptyMessage /> with default props should match the snapshot 1`] = `
<div
  className="col-12 dynamic-table-empty-message"
>
  There are no widgets matching the criteria.
</div>
`;
